pkg/virt-controller/network/hotplug.go

```diff
@@ -69,6 +69,16 @@ func ApplyDynamicIfaceRequestOnVMI(vm *v1.VirtualMachine, vmi *v1.VirtualMachine
 			vmiIface.State = v1.InterfaceStateAbsent
 		}
 	}
+	// Also unplug VMI interfaces that don't exist in VM spec
+	vmSpecCopy := vm.Spec.DeepCopy()
+	vmIndexedInterfaces := vmispec.IndexInterfaceSpecByName(vmSpecCopy.Template.Spec.Domain.Devices.Interfaces)
+	for _, vmiIface := range vmiSpecCopy.Domain.Devices.Interfaces {
+		_, existsInVMSpec := vmIndexedInterfaces[vmiIface.Name]
+		if !existsInVMSpec {
+			vmiIface := vmispec.LookupInterfaceByName(vmiSpecCopy.Domain.Devices.Interfaces, vmiIface.Name)
+			vmiIface.State = v1.InterfaceStateAbsent
+		}
+	}
```

Comment:
This opens up a second method to unplug a network interface.

I am unsure if it makes sense to do it.
We could argue that if someone deletes an interface, it means that it can take effect only on next boot.

But please try to reason for this on the commit message and PR description.